src/material/table/table-data-source.ts

```diff
@@ -27,20 +28,16 @@ import {map} from 'rxjs/operators';
  */
 const MAX_SAFE_INTEGER = 9007199254740991;
 
-/**
- * Data source that accepts a client-side data array and includes native support of filtering,
- * sorting (using MatSort), and pagination (using MatPaginator).
- *
- * Allows for sort customization by overriding sortingDataAccessor, which defines how data
- * properties are accessed. Also allows for filter customization by overriding filterTermAccessor,
- * which defines how row data is converted to a string for filter matching.
- *
- * **Note:** This class is meant to be a simple data source to help you get started. As such
- * it isn't equipped to handle some more advanced cases like robust i18n support or server-side
- * interactions. If your app needs to support more advanced use cases, consider implementing your
- * own `DataSource`.
- */
-export class MatTableDataSource<T> extends DataSource<T> {
+interface Paginator {
+  page: EventEmitter<PageEvent>;
```

Comment:
Maybe we should make this a `Subject` so it's Angular-agnostic? `EventEmitter` extends `Subject` already.

Reply:
Sure, I don't think its super important though, this is just an internal thing. The API we expose to users is still `MatPaginator`